scipy/stats/_discrete_distns.py

```diff
@@ -1049,6 +1057,79 @@ def _munp(self, n, a):
 zipf = zipf_gen(a=1, name='zipf', longname='A Zipf')
 
 
+def _gen_harmonic(n, a=1):
+    """Generalized harmonic number"""
+    # See https://en.wikipedia.org/wiki/Harmonic_number; search for "hurwitz"
+    return zeta(a, 1) - zeta(a, n+1)
+
+
+class zipfian_gen(rv_discrete):
+    r"""A Zipfian discrete random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability mass function for `zipfian` is:
+
+    .. math::
+
+        f(k, a, n) = \frac{1}{H_{n,a} k^a}
+
+    for :math:`k \ge 1`, :math:`a \ge 0`, :math:`n \in {1, 2, 3, \dots}`.
+
+    `zipfian` takes :math:`a` and :math:`n` as shape parameters.
+    :math:`H_{n,a}` is the :math:`n^{\mbox{th}}` generalized harmonic number
+    of order :math:`a`.
+
+    The Zipfian distribution converges to the Zipf (zeta) distribution as
+    :math:`n \rightarrow \infty`.
+
+    %(after_notes)s
+
+    References
+    ----------
+    "Zipf's Law", Wikipedia, https://en.wikipedia.org/wiki/Zipf's_law
+
+    %(example)s
+
+    """
+    def _argcheck(self, a, n):
+        return (a >= 0) & (np.asarray(n) == np.asarray(n, dtype=int))
```

Comment:
Asarrays not needed? IIRC all is converted to arrays when _argcheck is called

Reply:
Yes I meant to check. Thanks for the reminder.

Reply:
Apparently not:
```
return (a >= 0) & (n == n.astype(int))
E   AttributeError: 'int' object has no attribute 'astype'
```

Reply:
This happens in the `moment` method. I think it is being fixed in gh-12197.

Reply:
@mdhaber should this PR wait for gh-12197 to be merged then?

Reply:
I don't think so. It's not really hurting anything to ensure it's an array.